tests/unit/design_document_tests.py

```diff
@@ -1035,5 +1035,50 @@ def test_get_search_index(self):
                 '{"store": true}); }\n}'}
         )
 
+    def test_fetch_rewrite_rule(self):
+        """
+        Ensure that the document fetch from the database returns the
+        DesignDocument format as expected when retrieving a design document
+        containing rewrite rules.
+        """
+        ddoc = DesignDocument(self.db, '_design/ddoc001')
+        ddoc['rewrites'] = [
+            {"from": "/old/topic",
+             "to": "/new/",
+             "method": "GET",
+             "query": {}
+             },
+            {"from": "/foo/:var",
+             "to": "/foo",
+             "method": "GET",
+             "query": {"v": "var"}
+             }
+        ]
+        self.assertIsInstance(ddoc.rewrites, list)
+        self.assertIsInstance(ddoc.rewrites[0], dict)
+        ddoc.save()
+        ddoc_remote = DesignDocument(self.db, '_design/ddoc001')
+        ddoc_remote.fetch()
+        self.assertIsInstance(ddoc_remote.rewrites, list)
+        self.assertIsInstance(ddoc_remote.rewrites[0], dict)
+        self.assertEqual(ddoc_remote, {
```

Comment:
Is it possible to do an assert on the final url to check that it gets rewritten correctly?